build.gradle

```diff
@@ -29,6 +29,21 @@ allprojects {
 
 apply plugin: 'java'
 
+/* SET PROJECT INFO for all artifacts */
+assert hasProperty('version')   // See version in gradle.properties.
+System.out.println("version=${version}")
+group = 'org.apache.edgent'
+
+def getDate() {
```

Comment:
This getDate function counts on the build system to append date to the build version. I think you can safely move it without side effects